rust/private/rustc.bzl

```diff
@@ -1822,10 +1818,10 @@ def _add_native_link_flags(args, dep_info, linkstamp_outs, ambiguous_libs, crate
 
     use_pic = _should_use_pic(cc_toolchain, feature_configuration, crate_type, compilation_mode)
 
-    if toolchain.os == "windows":
+    if toolchain.target_os == "windows":
```

Comment:
I could be wrong but I think this section has a mixed use of the "os". The formatting of the link flags would be dependent on what tools are being used (MSVC's `link.exe` vs `gcc`). This may be something to address in https://github.com/bazelbuild/rules_rust/pull/1958 but worth considering in this change as well.